Fix handling of multi instance resources with one instance

nordic-mare opened the pull request:
This will fix a bug which occurs when a multi instance resource is created with only a single instance. Previously this was treated as a single instance resource and not a multi instance resource with one instance

Signed-off-by: nordic-mare <markus.rekdal@nordicsemi.no>

rlubos reviewed (commented):
Looks good, thanks.

Please update the commit message a bit though, add `net: lwm2m: ` prefix to the commit title, and some body explaining the reasoning behind the change.

rlubos commented:
One more issue with the commit message - it's required to use full name for the commit author and the signoff entry. Please update your `user.name` git config, and then `git commit --amend --no-edit --reset-author -s` (note you'll need to remove the previous entry manually).

nordic-mare committed:
net: lwm2m: Fix handling of multi instance resources with one instance

This will fix a bug caused by creating a multi instance resource with
only a single resource. Previously this was treated as a single instance
resource. This is now properly treated as a multi instance resource with
one instance
Signed-off-by: Markus Rekdal <markus.rekdal@nordicsemi.no>
